lib/orm.php

```diff
@@ -2541,4 +2589,124 @@ public function __unset( $key ) {
+	 *
+	 * @return bool
+	 */
+	private static function has_persistent_cache() {
+		if ( self::$has_persistent_cache === false || self::$has_persistent_cache === true ) {
+			return self::$has_persistent_cache;
+		}
+
+		// Check if the site is using an external object cache.
+		if ( \function_exists( 'wp_using_ext_object_cache' )
+			&& \wp_using_ext_object_cache()
+		) {
+			self::$has_persistent_cache = true;
+			return true;
+		}
+
+		// Defensive coding - cover all bases.
+		// This will prevent errors below if WP_CONTENT_DIR is not defined.
+		if ( ! defined( 'WP_CONTENT_DIR' ) ) {
+			return false;
+		}
+
+		// Check if the site is using a persistent drop-in (object-cache.php).
+		// This _should_ be covered by the `wp_using_ext_object_cache()` check,
+		// but sometimes isn't depending on the timing of the cache.
+		if ( \file_exists( WP_CONTENT_DIR . '/object-cache.php' ) ) {
+			self::$has_persistent_cache = true;
+			return true;
+		}
+
+		// No persistent caching was detected.
+		self::$has_persistent_cache = false;
+		return false;
+	}
+
+	/**
+	 * Get a cache.
+	 *
+	 * @param string $key      Cache key.
+	 * @param mixed  $group_id Cache group ID.
+	 *
+	 * @return mixed The cache value.
+	 */
+	public static function get_cache( $key, $group_id = '' ) {
+		// Bail early if the site doesnt have persistent caching.
+		if ( ! self::has_persistent_cache() ) {
+			return false;
+		}
+		$cache_group = self::get_cache_group( $group_id );
+		return \wp_cache_get( $key, $cache_group );
+	}
+
+	/**
+	 * Set a cache.
+	 *
+	 * @param string $key      Cache key.
+	 * @param mixed  $value    Cache value.
+	 * @param mixed  $group_id Cache group ID.
+	 *
+	 * @return void
+	 */
+	private static function set_cache( $key, $value, $group_id = '' ) {
+		// Bail early if the site doesnt have persistent caching.
+		if ( ! self::has_persistent_cache() ) {
+			return;
+		}
+		/**
+		 * Set the cache, with a timeout of 10 minutes.
+		 *
+		 * The 10-minute is long enough to significantly improve performance,
+		 * but short enough to not cause issues with stale data.
+		 *
+		 * Individual cache-groups are purged when a record is updated or deleted.
+		 */
+		\wp_cache_set( $key, $value, self::get_cache_group( $group_id ), ( 10 * MINUTE_IN_SECONDS ) );
+	}
+
+	/**
+	 * Flushes the cache for the given group IDs.
+	 *
+	 * @param mixed $group_id Cache group ID.
+	 *
+	 * @return void
+	 */
+	public static function flush_group_caches( $group_id = '' ) {
+		// Bail early if the site doesnt have persistent caching.
+		if ( ! self::has_persistent_cache() ) {
+			return;
+		}
+		// Bail early if the cache doesn't support group-flushing.
+		if ( \function_exists( 'wp_cache_supports' ) && ! \wp_cache_supports( 'flush_group' ) ) {
+			return;
+		}
+		// Bail early if the `wp_cache_flush_group` function doesn't exist.
+		if ( ! \function_exists( 'wp_cache_flush_group' ) ) {
+			return;
+		}
+		\wp_cache_flush_group( self::get_cache_group( $group_id ) );
+	}
```

Comment:
wp_cache_flush_group is a very new feature of object caching and is not very well supported. their needs to be a fallback here.

Reply:
Why validate everythings through?